src/gui/qgsrasterlayersaveasdialog.cpp

```diff
@@ -963,6 +965,21 @@ void QgsRasterLayerSaveAsDialog::accept()
     return;
   }
 
+  if ( QgsMapLayerUtils::isOpenStreetMapLayer( mRasterLayer ) )
+  {
+    const int nbTilesWidth = std::ceil( nColumns() / 256 );
+    const int nbTilesHeight = std::ceil( nRows() / 256 );
+    int64_t totalTiles = static_cast<int64_t>( nbTilesWidth ) * nbTilesHeight;
+
+    if ( totalTiles > 5000 )
```

Comment:
shouldn't we have that 5000 constant somewhere since it is used in multiple places?